client/src/signup/Signup.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import axios from 'axios';
+
+class Signup extends React.Component {
+  
+  state = {
+    username: '',
+    password: '',
+    department: ''
+  }
+
+  onInputChange = event =>  {
+    const { name, value } = event.target;
+    this.setState({ 
+      [name]: value
+    })
+  }
+
+  onSubmit = event => {
+    console.log(this.state)
+    event.preventDefault();
+    const endpoint = "http://localhost:9090/api/register"
+    
+    axios
+      .post(endpoint, this.state)
+      .then(res => {
+        console.log(res.data)
+        localStorage.setItem('jwt', res.data.token)
+        this.props.history.push('/users')
+      })
+      .catch(err => {
+        console.error(err)
+      })
+  }
+
+  render(){
+    return (
+      <>
+        <h2>Sign Up</h2>
+        <form onSubmit={this.onSubmit}>
+          <label htmlFor="username">Username</label>
+          <input 
+            name="username"
+            type="text"
+            id="username"
+            value={this.state.username}
+            onChange={this.onInputChange}
+          />
+          <label htmlFor="password">Password</label>
+          <input 
+            name="password"
+            type="text"
+            id="password"
+            value={this.state.password}
+            onChange={this.onInputChange}
+          />
+          <label htmlFor="department">Department</label>
+          <input 
+            name="department"
+            type="text"
+            id="department"
+            value={this.state.department}
+            onChange={this.onInputChange}
+          />
```

Comment:
These inputs seem repetitive; do you think you could make a component for them? Or, for the sake of an example, a method:
```javascript
const makeInput = (name, type = 'text') => (
  <>
    <label htmlFor={name}>{name[0].toUpperCase() + name.slice(1)}</label>
    <input
      name={name}
      type={type}
      id={name}
      value={this.state[name]}
      onChange={this.onInputChange}
    />
  </>
);
```
and the usage:
```javascript
<form onSubmit={this.onSubmit}>
  {this.makeInput('username')}
  {this.makeInput('password')}
  {this.makeInput('department')}
  <button type="submit">Register</button>
</form>
```